### DISCUSSION TITLE
Disable "ServiceAuthorize" for one method inside a Controller

### DISCUSSION THREAD

**Discussion by ghost** [serenity-is/Serenity#5887] [Help] [answered]
Hello, my XYZEndpoint.cs file has attribute "ServiceAuthorize(typeof(MyRow))" that applies throughout the entire controller. Is there a way to disable it for one method? 
I tried following lines above my method but to no avail: 
```
ServiceAuthorize()
ServiceAuthorize(false)
```
The reason why I want to do this is that I am using a different kind of authorization just for this specific method

**Comment by edwardch** [comment]
Not sure if it's the right way, but you could try to create a new controller in the same XYZEndpoint.cs file like so:
```cs
[Route("Services/XyzModule/Xyz/[action]")]
[ConnectionKey(typeof(MyRow))]
[ServiceAuthorize("*")] // the * should work, if not try remove this line completely
public class NewXyzController : ServiceEndpoint
{
    // your code here
}

// Existing code
[Route("Services/XyzModule/Xyz/[action]")]
[ConnectionKey(typeof(MyRow)), ServiceAuthorize(typeof(MyRow))]
public class XyzController : ServiceEndpoint
{
    [HttpPost, AuthorizeCreate(typeof(MyRow))]
    public SaveResponse Create(IUnitOfWork uow, SaveRequest<MyRow> request)
    {
        return new MyRepository().Create(uow, request);
    }
    // ...
    // other methods
    // ...
}    
```

**Reply by ghost** [reply]
Good idea sir, thank you

**Comment by VictorTomaili** [comment, marked as answer]
When you write `ServiceAuthorizeAttribute` on class and action, it checks both of permission with `and`. If you want separate permissions for one action, you must remove this attribute from the class then write on action instead. In the default scenario, you already have `AuthorizeCreate`, `AuthorizeUpdate` and `AuthorizeDelete` attributes on actions. You can remove `ServiceAuthorize(typeof(MyRow))` from class and add them on `Retrieve` and `List`. 
For non-logged-in access, you can add `ServiceAuthorize("*")` for your public action. 
For logged-in access, you can add `ServiceAuthorize("?")` for your action. 

In all scenarios, if you have the `ServiceAuthorize` attribute on the controller, this controller permission is still required (and it's checks read permission on the row)

**Reply by ghost** [reply]
Great explanation, thank you.

**Reply by ghost** [reply]
Hey @VictorTomaili , what about the native Authorize attribute? Does it influence Serenity's ServiceAuthorize attribute?

I have ServiceAuthorize(typeof(MyRow)) on my Controller. I also put Authorize["JWToken"] above my API method. 
To my surprise, calling this method works even when not logged-in.
